docs/structured.md

```diff
@@ -0,0 +1,207 @@
+
+##Goals##
+
+Add support for the following entities: <i>namespaces, tables, columns, and
+indexes</i>. These notions are chosen for their similarity to <i>databases, tables,
+columns, and indexes</i> in relational databases, but they are not intended to be
+implementations thereof. Each entity will support:
+
+- Creation: of a namespace, table, column in O(1).
+
+- Renaming: in O(1) Most configuration changes in O(1): A change in permissions
+  for instance, will be in O(1) time. Adding a new index will take O(n) time,
+where n is the number of rows in the table.
+
+- Deletion: in O(n) where n is the number of subordinate entities (tables for a
+  namespace, rows for a table).
+
+The client API will be changed to use tabular based addressing instead of
+global-key addressing. Non-primary-keyed queries will be supported through the use of secondary indexes.
+
+##Non-Goals##
+
+For now we don't intend developing the following:
+
+- Add support for triggers, stored-procedures, and integrity constraints
+
+- Add the notion of column types beyond <i>a sequence of bytes</i>. Add support for
+  unbounded sub-collections of columns.
+
+- Add support for non-tabular storage. This might be supported in the future.
+
+- Add support for column locality-groups to improve data layout for performance.
+
+## Design ##
+
+We support high level entities like Namespace, Table, Index, and Column.
+{Namespace,Table}Descriptor stores the entity metadata for {Namespace,Table}.
+
+<pre>
+ <code>NamespaceDescriptor {
+         NamespaceID = … ,
+         Permissions = …,
+         ...}</code>
+
+ <code>TableDescriptor {
+         TableID = …,
+         Columns = [{ ColumnID = …, Name = ... }, ...],
+         Indexes = [{ IndexID = …, Name = …, ColumnIDs = [ … ]}, ...],
+         Permissions = …
+         NextFreeColumnId = …
+         NextFreeIndexID = ...,
+         ...}</code>
+</pre>
+
+In order to support fast renaming, we indirect entity access through a
+hierarchical name-to-identifier mapping. NamespaceID and TableID are
+allocated from the same global ID space. ColumnID and IndexID are local
+to each Table.
+
+To simplify our implementation, our Tables will require the presence of a
+primary key. Follow-up work may relax this requirement. An investigation of use
+cases not requiring a primary key is required to specify this work. A cell in
+the Table can be addressed for a particular Column and Row.
+
+Initially, all the Table metadata in a cluster will be distributed by gossip and
+cached by each cockroach node. To reduce contention on the Table metadata, a
+database transaction on the data within a Table will only contain the data being
+addressed, and will not include a database read of the Table metadata. Care will
+be taken to not expose these implementation details to the user.
+
+##Addressing: Anatomy of a key##
+
+The “/” separators used below are shorthand for ordered encoding of the
+separated values.
+
+*Global metadata keys*
```

Comment:
i think you want to use a heading here (Rather than emphasis):
###Global metadata keys